src/plugins/profiling/common/flamegraph.ts

```diff
@@ -55,120 +152,116 @@ export class FlameGraph {
   stackframes: Map<StackFrameID, StackFrame>;
   executables: Map<FileID, Executable>;
 
-  private readonly logger: Logger;
-
   constructor(
     sampleRate: number,
     totalCount: number,
     events: Map<StackTraceID, number>,
     stackTraces: Map<StackTraceID, StackTrace>,
     stackFrames: Map<StackFrameID, StackFrame>,
-    executables: Map<FileID, Executable>,
-    logger: Logger
+    executables: Map<FileID, Executable>
   ) {
     this.sampleRate = sampleRate;
     this.totalCount = totalCount;
     this.events = events;
     this.stacktraces = stackTraces;
     this.stackframes = stackFrames;
     this.executables = executables;
-    this.logger = logger;
   }
 
-  private getExeFileName(exe: any, type: number) {
-    if (exe?.FileName === undefined) {
-      return '';
-    }
-    if (exe.FileName !== '') {
-      return exe.FileName;
-    }
-    switch (type) {
-      case 0:
-        return '<unsymbolized frame>';
-      case 1:
-        return 'Python';
-      case 2:
-        return 'PHP';
-      case 3:
-        return 'Native';
-      case 4:
-        return 'Kernel';
-      case 5:
-        return 'JVM/Hotspot';
-      case 6:
-        return 'Ruby';
-      case 7:
-        return 'Perl';
-      case 8:
-        return 'JavaScript';
-      default:
-        return '';
+  // createColumnarCallerCallee flattens the intermediate representation of the diagram
+  // into a columnar format that is more compact than JSON. This representation will later
+  // need to be normalized into the response ultimately consumed by the flamegraph.
+  private createColumnarCallerCallee(root: CallerCalleeNode): ColumnarCallerCallee {
```

Comment:
Great design choice IMO 👍🏼 
Keeping the data conversion server-side was also suggested by the data viz team IIRC